backend/accounts/transaction.go

```diff
@@ -54,40 +55,51 @@ type AddressAndAmount struct {
 	Ours bool
 }
 
-// Transaction models a transaction with common transaction info.
-type Transaction interface {
+// TransactionData holds transaction data to be shown to the user. It is as coin-agnostic as
+// possible, but contains some fields that are only used by certain coins.
+type TransactionData struct {
 	// Fee is nil for a receiving tx. The fee is only displayed (and relevant) when sending funds
 	// from the wallet.
-	Fee() *coin.Amount
-
+	Fee *coin.Amount
```

Comment:
I'd remove pointer. It's like a `type Point struct {x, y int}` or `time.Time`. 

```go
// Amount represents an amount in the smallest coin unit (e.g. satoshi).
type Amount struct {
	n *big.Int
}
```

So, we have pointer to `Amount` which has only one field, pointer to `big.Int`. If `Fee` is not `nil` but `Fee.n` is, that's kind of a weird and invalid state. Maybe a bit simplified version would be:

```go
type Amount big.Int
Fee *coin.Amount
```

then a Fee pointer field makes a bit more sense imho. Probably not in this PR though.

Reply:
It's a pointer because it can be missing / nil.

The Amount field element should never be nil. 

That is a problem of Go: using pointers to save space vs. indicating that it is not present. They interfere with each other :/

Reply:
Will check in a new PR if I can improve it.